### PR TITLE
Add rectangles as a primitive asset.

### LINKED ISSUE #566: Create a Rectangle asset to go with the Square.
We don't have an easy way to generate simply rectangular sprites, this would assist with that.

### PR DESCRIPTION
**Pull request by pathunstrom**
Work to support rectangular shape assets.

Resolves #566

### CODE REVIEW

**Review by AstraLuma — CHANGES_REQUESTED** [review]
Can you add a viztest for this?

File: ppb/assets.py [hunk 1]
```diff
@@ -92,33 +115,52 @@ def _draw_shape(self, renderer, rgb, **_):
         )
 
 
+def Square(r, g, b):
```
Comment (AstraLuma) [inline ppb/assets.py]: Should we turn this into classes or just leave them as functions?

**Comment by pathunstrom** [comment]
> Can you add a viztest for this?

Will do, I considered just, adding to our rectangle_sprite viz test, but I figure the example is also a visual test so can just. . . move it over.

File: ppb/assets.py (hunk 1 above)
Comment (pathunstrom) [reply ppb/assets.py]: Either way is fine with me.

I just wanted to keep this change backwards compatible and I want to document that it's basically just a wrapper of the "real" implementation.

File: ppb/assets.py [hunk 2]
```diff
@@ -92,33 +115,56 @@ def _draw_shape(self, renderer, rgb, **_):
         )
 
 
+class Square(Rectangle):
+    """
+    A constructor for :class:`~ppb.Rectangle` that produces a square image.
+    """
+
+    def __init__(self, r, g, b):
+        super().__init__(r, g, b)
```
Comment (AstraLuma) [inline ppb/assets.py]:
```suggestion
    def __init__(self, r, g, b):
        # This cuts out the aspect_ratio parameter
        super().__init__(r, g, b)
```

File: ppb/assets.py [hunk 3]
```diff
@@ -92,33 +115,56 @@ def _draw_shape(self, renderer, rgb, **_):
...
             *rgb, 255,
             _check_error=lambda rv: rv < 0
         )
+
+
+class Circle(Ellipse):
+    """A convenience constructor for :class:`~ppb.Ellipse` that is a perfect circle."""
+
+    def __init__(self, r, g, b):
+        super().__init__(r, g, b)
```
Comment (AstraLuma) [inline ppb/assets.py]:
```suggestion
    def __init__(self, r, g, b):
        # This cuts out the aspect_ratio parameter
        super().__init__(r, g, b)
```

**Review by AstraLuma — APPROVED** [review]
Looks good. I would strongly suggest incorporating the suggested comments, though.

**Pull request merged**